File: demo-java/src/main/java/com/cappasity/demojava/ModelFragment.java
//--------------------------------------------------------------------------------------
// Cappasity technology / Cappasity 3D Scanning software solutions. U.S. Patent Pending.
//
// CONTENT IS PROHIBITED FROM USAGE. CAPPASITY INC. CONFIDENTIAL.
//
// Copyright (C) Cappasity Inc. 2013-2020. All rights reserved.
//--------------------------------------------------------------------------------------

package com.cappasity.demojava;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.cappasity.framework.CappasityModel;
import com.cappasity.framework.CappasityModelView;
import com.cappasity.framework.CappasityModelViewParams;

public class ModelFragment extends Fragment {

    static String ARG_MODEL = "model";

    private CappasityModel model;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle arguments = getArguments();
        if (arguments != null) {
            model = arguments.getParcelable(ARG_MODEL);
        }
        ((AppCompatActivity) getActivity()).getSupportActionBar().hide();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View root;
        if (model != null) {
            root = inflater.inflate(R.layout.fragment_model, container, false);
            final CappasityModelView modelView = root.findViewById(R.id.modelView);
            final ProgressBar progressBar = root.findViewById(R.id.pb);
            final CappasityModelViewParams params = new CappasityModelViewParams.Builder()
                    .autoRun(true)
                    .build();
            modelView.loadModel(model, params);
            modelView.setOnModelLoadListener(new CappasityModelView.OnModelLoadListener() {
                @Override
                public void onModelLoad() {
                    modelView.setVisibility(View.VISIBLE);
                    progressBar.setVisibility(View.GONE);
                }
            });
            return root;

        }
        root = inflater.inflate(R.layout.fragment_model_empty, container, false);
        return root;
    }

}

File: demo-kotlin/src/main/java/com/cappasity/demokotlin/ModelFragment.kt
//--------------------------------------------------------------------------------------
// Cappasity technology / Cappasity 3D Scanning software solutions. U.S. Patent Pending.
//
// CONTENT IS PROHIBITED FROM USAGE. CAPPASITY INC. CONFIDENTIAL.
//
// Copyright (C) Cappasity Inc. 2013-2020. All rights reserved.
//--------------------------------------------------------------------------------------

package com.cappasity.demokotlin

import android.os.Bundle
import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.cappasity.framework.CappasityModel
import com.cappasity.framework.CappasityModelViewParams
import kotlinx.android.synthetic.main.fragment_model.*


class ModelFragment : Fragment(R.layout.fragment_model) {

    companion object {
        const val ARG_MODEL = "model"
    }

    private val model: CappasityModel
        get() = arguments?.getParcelable(ARG_MODEL)!!


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        (activity as AppCompatActivity?)!!.supportActionBar!!.hide()
        val modelViewParams = CappasityModelViewParams(
            autoRun = true
        )
        modelView.loadModel(
            model,
            modelViewParams
        )

        modelView.setOnModelLoadListener {
            modelView.visibility = VISIBLE
            pb.visibility = GONE
        }
    }

}
